app/services/fees/create_recurring_pay_in_advance_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Fees
+  class CreateRecurringPayInAdvanceService < BaseService
+    def initialize(subscriptions:, billing_at:)
+      @subscriptions = subscriptions
+      @billing_at = billing_at
+
+      super
+    end
+
+    def call
+      plan_ids = subscriptions.select(&:active?).map(&:plan_id).uniq
+
+      Charge.joins(:billable_metric)
+        .where(plan_id: plan_ids, pay_in_advance: true, invoiceable: false)
+        .where(billable_metrics: {recurring: true})
+        .find_each do |charge|
+        last_fee = charge.fees.order(created_at: :desc).first
+        event = Event.find_by(id: last_fee&.pay_in_advance_event_id)
```

Comment:
We will have to adapt this condition for in advance handling in clickhouse

Reply:
I'm not too sure what you would need to change, I think all of this is happening in Rails

Reply:
do we need to change something on our side? or will this be done in another pr related to the clubhouse?

Reply:
**Let's handle it with clickhouse.**

In Clickhouse, events does not have a unique ID like the one in PG, and so we will not be able to link in advance events to fees like we are doing here

Reply:
Ah! I wasn't aware of that. The `Fees::CreatePayInAdvanceService` is really based on the event so we'll definitely need to pass it an event.

It's probably possible to refactor the service not depend on event but directly on some attributes (subscriptions, customers, date...) but it's not easy. Maybe easier than other solutions tho.

Reply:
Excellent news @vincent-pochet. This is not at all how we want this to work 🙈 The actual version won't need the event.